x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/request/anthropic/AnthropicChatCompletionRequestEntity.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.external.request.anthropic;
+
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.inference.services.anthropic.completion.AnthropicChatCompletionServiceSettings;
+import org.elasticsearch.xpack.inference.services.anthropic.completion.AnthropicChatCompletionTaskSettings;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+public class AnthropicChatCompletionRequestEntity implements ToXContentObject {
+
+    private static final String MESSAGES_FIELD = "messages";
+    private static final String MODEL_FIELD = "model";
+
+    private static final String ROLE_FIELD = "role";
+    private static final String USER_FIELD = "user";
+    private static final String CONTENT_FIELD = "content";
+    private static final String MAX_TOKENS_FIELD = "max_tokens";
+    private static final String TEMPERATURE_FIELD = "temperature";
+    private static final String TOP_P_FIELD = "top_p";
+    private static final String TOP_K_FIELD = "top_k";
+
+    private final List<String> messages;
+    private final AnthropicChatCompletionServiceSettings serviceSettings;
+    private final AnthropicChatCompletionTaskSettings taskSettings;
+
+    public AnthropicChatCompletionRequestEntity(
+        List<String> messages,
+        AnthropicChatCompletionServiceSettings serviceSettings,
+        AnthropicChatCompletionTaskSettings taskSettings
+    ) {
+        this.messages = Objects.requireNonNull(messages);
+        this.serviceSettings = Objects.requireNonNull(serviceSettings);
+        this.taskSettings = Objects.requireNonNull(taskSettings);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+
+        builder.startArray(MESSAGES_FIELD);
+        {
+            for (String message : messages) {
+                builder.startObject();
+
+                {
+                    builder.field(ROLE_FIELD, USER_FIELD);
```

Comment:
is this right? the field value is a field string?

Reply:
Yeah I'm hardcoding the role to `user` here. In theory this could be specified but for all our other chat completion implementations we've hard coded it so I figured probably best to keep it consistent for now until we decided to provide that capability and then change it for all of them.

Reply:
That makes sense. Definitely a nit, but I would prefer if we used a different variable name for `user` then, since this isn't actually a field name in this context. Even just creating a constant `String USER_VALUE = USER_FIELD` I feel like would be more clear? Up to you though.

Reply:
Oh I see, yeah you're right the name doesn't make sense, I'll update 👍